libvast/include/vast/plugin.hpp

```diff
@@ -313,7 +313,8 @@ class store_plugin : public virtual store_actor_plugin {
 
   /// Create a store for active partitions.
   [[nodiscard]] virtual caf::expected<std::unique_ptr<active_store>>
-  make_active_store() const = 0;
+  make_active_store(
+    const caf::dictionary<caf::config_value>& vast_config) const = 0;
```

Comment:
1. `caf::dictionary<caf::config_value>` is usually called `caf::settings` (which is an alias for the former, but much more readable)
2. Do we really want to pass in _all_ settings here instead of just the one that is relevant? One of the goals of the `store_plugin` was to make it simple to use over the `store_actor_plugin`.

Reply:
The main argument for passing in all settings is that we don't know which settings might be relevant for a plugin. E.g., if a plugin doesn't use zstd, but a different one, we'd have to extend the interface, and touch all plugin implementations. On the other hand, parquet could possibly use the global `table_slice_size` for producing properly sized chunks when reading parquet, but currently uses its own setting for this.

Reply:
I've see the passing-in-all-settings pattern across languages quite frequently. I think it works, actually, given that the configuration is often considered public global state.